docs/sphinx/user_guide/feature/loop_basic.rst

```diff
@@ -250,6 +250,7 @@ statement types:
   * ``RAJA::statement::CudaKernel< EnclosedStatements>`` launches 'EnclosedStatements' as a CUDA kernel; e.g., a loop nest where iteration space of each loop level are associated to threads and/or thread blocks. 
   * ``RAJA::statement::CudaSyncThreads`` provides CUDA '__syncthreads' barrier; a similar thread barrier for OpenMP will be added soon.
   * ``RAJA::statement::Hyperplane< ArgId, HpExecPolicy, ArgList<...>, ExecPolicy, EnclosedStatements >`` provides a hyperplane iteration pattern over multiple indices.
+  * ``RAJA::statement::Reduce< ReducePolicy, Operator, ParamId, EnclosedStatements >`` reduces a value in parameter ParamId across threads to a single thread. After reduction, executes 'EnclosedStatements' on thread that recieved the final reduced value.
```

Comment:
Should it be noted that thread zero always receives the final reduced value?

Reply:
I don't think so...  I wanted to abstract that notion from the user.  The enclosed statements only get executed on the thread that got the reduced value... which is all they should care about (i hope).  Plus, I want to leave open the ability to add a cuda_warp_reduce, which would mean it's not thread zero.  Also, I have no idea what we're going to do with OpenMP (if anything).

Reply:
Also, I would like to eventually add a ReduceAll, which wouldn't have enclosed values, and all threads would get the reduced value.

Reply:
Sounds wonderful!

Reply:
This makes me wonder, is there any way for a user to reason about the value in that param in a lambda immediately following the reduce at the same nesting level? It sounds like not?

Reply:
The way it's implemented, no.  I'm open to figuring out something better, like only modifying the param on the thread that gets the final value?!?

Reply:
Maybe? I think we'll have to play with it and see what makes the most sense as users see it.  We may even be best off having the broadcast version be the default, or make it so that when you have a nested loop with a reduction in the inner loop the thread executing in the outer loop context always gets it?